### PR TITLE
Apply backup/restore CRDs if available

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Deploys `CheClusterBackup` and `CheClusterRestore` CRDs if available in templates.

### How to test this PR?
Just deploy Che

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by tolusha** [comment]
What about update task?

File: src/tasks/installers/operator.ts [hunk 1]
```diff
@@ -184,6 +186,29 @@ export class OperatorTasks {
           }
         }
       },
+      {
+        title: 'Create backup and restore CRDs',
+        task: async (ctx: any, task: any) => {
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:191]: if file does not exist I prefer `${task.title}...skipped.` like with other crd

File: src/tasks/installers/operator.ts [hunk 2]
```diff
@@ -356,6 +381,34 @@ export class OperatorTasks {
...
         }
       },
+      {
+        title: 'Updating backup and restore CRDs',
+        task: async (ctx: any, task: any) => {
+          const existedBackupCRD = await kube.getCrd(this.cheClusterBackupCrd)
+          const newBackupCRDPath = path.join(ctx.resourcesPath, 'crds', 'org.eclipse.che_checlusterbackups_crd.yaml')
+          if (existedBackupCRD) {
+            if (!existedBackupCRD.metadata || !existedBackupCRD.metadata.resourceVersion) {
+              throw new Error(`Fetched CRD ${this.cheClusterCrd} without resource version`)
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:391]: copy/paste.. it is not checluster crd

File: src/tasks/installers/operator.ts [hunk 3]
```diff
@@ -356,6 +381,34 @@ export class OperatorTasks {
...
+            await kube.replaceCrdFromFile(newBackupCRDPath, existedBackupCRD.metadata.resourceVersion)
+          } else {
+            await kube.createCrdFromFile(newBackupCRDPath)
+          }
+
+          const existedRestoreCRD = await kube.getCrd(this.cheClusterRestoreCrd)
+          const newRestoreCRDPath = path.join(ctx.resourcesPath, 'crds', 'org.eclipse.che_checlusterrestores_crd.yaml')
+          if (existedRestoreCRD) {
+            if (!existedRestoreCRD.metadata || !existedRestoreCRD.metadata.resourceVersion) {
+              throw new Error(`Fetched CRD ${this.cheClusterCrd} without resource version`)
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:402]: copy/paste.. it is not checluster crd

File: src/tasks/installers/operator.ts [hunk 4]
```diff
@@ -503,6 +556,19 @@ export class OperatorTasks {
         }
       }
     },
+    {
+      title: 'Delete backup and restore CRDs',
```
Comment (tolusha) [inline src/tasks/installers/operator.ts:560]: I think it can be merged with the task above.

File: src/tasks/installers/operator.ts (hunk 1 above)
Comment (mmorhun) [reply src/tasks/installers/operator.ts:191]: ok

File: src/tasks/installers/operator.ts (hunk 2 above)
Comment (mmorhun) [reply src/tasks/installers/operator.ts:391]: Thanks for spotting

**Review by tolusha — APPROVED** [review]

**Pull request merged**